backend/init/init.go

```diff
@@ -38,6 +39,7 @@ func init() {
 		"consul": func() backend.Backend { return backendconsul.New() },
 		"inmem":  func() backend.Backend { return backendinmem.New() },
 		"s3":     func() backend.Backend { return backendS3.New() },
+		"azure":  func() backend.Backend { return backendAzure.New() },
```

Comment:
I think we can add `"azurerm"` with the same init function, leaving the `"azure"` name in there and document it as deprecated for now. 

I'm not sure there's an easy way to wrap `backendAzure.New()` to trigger a deprecation warning, but I can look into it.